packages/stylist-core/src/registry.rs

```diff
@@ -68,25 +61,23 @@ mod tests {
         }
 
         assert_eq!(style_a.get_style_str(), style_b.get_style_str());
-        Ok(())
     }
 
     #[test]
-    fn test_duplicate_style_different_prefix() -> Result<()> {
+    fn test_duplicate_style_different_prefix() {
         init();
 
-        let style_a = Style::create("element-a", r#"color: red;"#)?;
-        let style_b = Style::create("element-b", r#"color: red;"#)?;
+        let style_a = Style::create_from_scopes("element-a", sample_scopes());
+        let style_b = Style::create_from_scopes("element-b", sample_scopes());
 
-        assert_ne!(style_a.get_style_str(), style_b.get_style_str());
-        Ok(())
+        assert_eq!(style_a.get_style_str(), style_b.get_style_str());
```

Comment:
Changed: equality in the StyleRegistry does *not* respect the classname. As this is a hint only, anyway, and an unpredictable rng string is appended; I don't see the point of caching by name.

One worry would be that users take the generated classname and insert additional css with some other method that then also effects deduplicated styles with the same scopes, but imo this should be pointed out in the documentation as bad practice and not be done. Rather one'd create an additional classname and add additional styles by combining multiple classes via the element's `class = "element-a element-b"` attribute.

Reply:
It used to be the class name of the Style, but I have changed it to be the prefix.

The first item in the tuple is the prefix ("element-a" or "stylist" by default), not the generated class name with entropy.
Yes, it’s purely visual, but somewhat important for debugging.

The purpose of the test is to test that the class prefix provided to Style::create is respected.
(Although, maybe testing the class name here is better.)

```rust
    // assert_ne!(".element-a-lCzxVWIt", ".element-b-WjX1zwQ1”);
    assert_ne!(element_a.get_class_name(), element_b.get_class_name());
```

If StyleKey is changed to save the parsed AST only, it will only use the prefix of the first generated stylesheet for all following stylesheets that has the same AST (both Styles will have a prefix of "element-a").

Reply:
If it's for debugging, I wonder if it could be better to maintain a list of original prefixes in debug mode and add them to a `data-` attribute of the generated style element? Just thought mounting multiple components for the same ast would be more wasteful.

Reply:
I cannot imagine the performance impact will be too much (Once cached). And there's no impact if you don't use it.

This PR is getting very large. Can we keep the current behaviour and address this in another round (with #11, as that also requires changes to mounting) if it's of your concern?

Reply:
Replaced with `assert_ne!(element_a.get_class_name(), element_b.get_class_name());` and added the string back to the StyleKey. Doing a copy here instead of storing a `Cow<...>`, compared to the cost of hashing and comparing the ast, that should be negligible.